compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -652,10 +652,22 @@ impl<'f> PerFunctionContext<'f> {
                 }
             }
             TerminatorInstruction::Return { return_values, .. } => {
+                // We want to fetch all aliases of each return value to be marked unknown as an array
+                // of references can potentially be aliased by those internal references.
+                let mut all_aliases = Vec::new();
+                for return_value in return_values {
+                    if let Some(expression) = references.expressions.get(return_value) {
+                        if let Some(aliases) = references.aliases.get(expression) {
+                            aliases.for_each(|alias| {
+                                all_aliases.push(alias);
+                            });
+                        }
+                    }
+                }
```

Comment:
Maybe this, and the one [here](https://github.com/noir-lang/noir/blob/8580c15e123c222f49dc55a674f08f026242a525/compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs#L527-L536) and [here](https://github.com/noir-lang/noir/blob/8580c15e123c222f49dc55a674f08f026242a525/compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs#L462-L464) could be merged into a helper function that takes a list value IDs to iterate and a lambda that tells what to do with the alias. Perhaps it could be arranged into further helpers to make it more intuitive to the casual reader to connect "anything which acts on a reference in these values" to gathering aliases.

Reply:
> It doesn't look like this handles the cases where expressions or aliases may return `None` - e.g. expressions with unknown aliases.

@jfecher I added that test. `v0` will still be included as an alias of v4, so we will not remove `store u1 1 at v0`.